config/main.py

```diff
@@ -5846,6 +5846,123 @@ def smoothing_interval(interval, rates_type):
 for plugin in helper.load_plugins(plugins):
     helper.register_plugin(plugin, config)
 
+#
+# 'subinterface' group ('config subinterface ...')
+#
+@config.group()
+@click.pass_context
+@click.option('-s', '--redis-unix-socket-path', help='unix socket path for redis connection')
+def subinterface(ctx, redis_unix_socket_path):
+    """subinterface-related configuration tasks"""
+    kwargs = {}
+    if redis_unix_socket_path:
+        kwargs['unix_socket_path'] = redis_unix_socket_path
+    config_db = ConfigDBConnector(**kwargs)
+    config_db.connect(wait_for_init=False)
+    ctx.obj = {'db': config_db}
+
+def subintf_vlan_check(config_db, parent_intf, vlan):
+    subintf_db = config_db.get_table('VLAN_SUB_INTERFACE')
+    subintf_names = [k for k in subintf_db if type(k) != tuple]
+    for subintf in subintf_names:
+        sub_intf_sep_idx = subintf.find(VLAN_SUB_INTERFACE_SEPARATOR)
+        if sub_intf_sep_idx == -1:
+            continue
+        if parent_intf == subintf[:sub_intf_sep_idx]:
```

Comment:
Can we please put these new Click commands in this PR description for easy reference? also, hope you are planning to update the Click config guide?

Reply:
Done.